docs/finders/advanced.mdx

```diff
@@ -81,6 +81,56 @@ You can also change the timeout ad-hoc:
 await $(#addComment).tap(findTimeout: Duration(seconds: 30));
 ```
 
+### You gotta pump it up! But which one to use?
+
+In Flutter, "pumping" means rendering frames to the screen.
+
+If there are no frames to pump, no animations are pending, which usually means
+that the next action during the test can be executed. It is an equivalent of
+what a human tester would do while testing an app - they would wait until the
+app's state stabilizes after they've done something. For example, they tap on a
+button and get redirected to another screen, but the data that will be shown
+there hasn't been loaded yet. In such a case, a human tester waits until a
+loader (or other animation) finishes. Pumping mechanism does exactly this - it
+renders consecutive frames on the screen. For how long, exactly? Usually, we
+want to pump frames as long as they come. That's what [`pumpAndSettle()`] does.
+
+`pumpAndSettle()` method is called by default inside all actions that can be
+performed while testing - tapping, scrolling, entering text, and so on. You can
+change that by setting the `settlePolicy` argument:
+
+```dart
+await $('Delete account').tap(settlePolicy: SettlePolicy.settle);
+await $('Confirm').tap(settlePolicy: SettlePolicy.pump);
+```
+
+`SettlePolicy` is an `enum` with 3 values. The default is `SettlePolicy.settle`
+but you can change it to `pump` or `trySettle`. While `settle` and `pump` simply
+refer to Flutter's built-in methods, `trySettle` is available only in Patrol.
+How is it different from other ones?
+
+`pumpAndTrySettle()` is pretty much like `pumpAndSettle()`, the only difference
+is that `pumpAndSettle()` throws an exception, if there were still new frames to
+render after sonme defined timeout, while `pumpAndTrySettle()` does not. That's
+why it has "`try`" in it's name.
+
```

Comment:
@jBorkowska I think it could be very useful for people to see how `SettlePolicy` values map to `pump()`, `pumpAndSettle()`, and `pumpAndTrySettle()`. Basically what's already in the docs of `SettlePolicy`, but here.

Otherwise LGTM.